openedx/core/djangoapps/site_configuration/management/commands/create_site_configuration.py

```diff
@@ -1,50 +0,0 @@
-"""
-Run by ansible to setup a single site configuration in sandbox environments
-"""
-import json
-import logging
-from textwrap import dedent
-
-from django.contrib.sites.models import Site
-from django.core.management.base import BaseCommand
-
-from openedx.core.djangoapps.site_configuration.models import SiteConfiguration
-
-LOG = logging.getLogger(__name__)
-
-
-class Command(BaseCommand):
-    """
-    Command to create a configuration for a single site.  If the site does not
-    already exist one will be created.
-
-    Example:
-    ./manage.py lms create_configuration uox.sandbox.edx.org
-      --configuration="{'COURSE_CATALOG_API_URL':'https://discovery-uox.sandbox.edx.org/api/v1'}"
-    """
-    help = dedent(__doc__).strip()
-
-    def add_arguments(self, parser):
-        parser.add_argument('domain')
-        parser.add_argument(
-            '--configuration',
-            type=json.loads,
-            help="Enter JSON site configuration",
-            required=False,
-            default=''
-        )
-
-    def handle(self, *args, **options):
-        domain = options['domain']
-        configuration = options['configuration']
-        site, created = Site.objects.get_or_create(
-            domain=domain,
-            name=domain,
-        )
-        if created:
-            LOG.info(u"Site does not exist. Created new site '{site_name}'".format(site_name=domain))
-        else:
-            LOG.info(u"Found existing site for '{site_name}'".format(site_name=domain))
-
-        LOG.info(u"Creating '{site_name}' SiteConfiguration".format(site_name=domain))
-        SiteConfiguration.objects.create(site=site, values=configuration, enabled=True)
```

Comment:
@lgp171188 This bit already replaced `SiteConfiguration` (good to keep a history), can we tweak this to allow both creating a new entry or updating a present one?
This way we can keep the current behaviour of this management command while adding a `update` parameter (that defaults to `False` and allows to update without creating a new configuration)

Reply:
@giovannicimolin, the `SiteConfiguration` model has a [one-to-one relationship](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/site_configuration/models.py#L32) with the `Site` model. So there can only be a single `SiteConfiguration` instance per site.

Also, naming the command `create_or_update_site_configuration` is more explicit (like the existing `create_or_update_site` management command) than adding an optional `--update` flag to a `create_site_configuration` command.